### PR TITLE
Group member functions in hp::FECollection for doxygen.

### PR DESCRIPTION
**Pull request by bangerth**
/rebuild

### CODE REVIEW

File: include/deal.II/fe/fe.h [hunk 1]
```diff
@@ -1270,7 +1270,7 @@ class FiniteElement : public Subscriptor, public FiniteElementData<dim>
   //@}
 
   /**
-   * @name Functions to support hp
+   * @name Functions to support hp adaptivity
```
Comment (marcfehling) [inline include/deal.II/fe/fe.h:1273]:
```suggestion
   * @name Functions to support hp-adaptivity
```

File: include/deal.II/fe/fe.h [hunk 2]
```diff
@@ -1312,7 +1312,7 @@ class FiniteElement : public Subscriptor, public FiniteElementData<dim>
 
 
   /**
-   * @name Functions to support hp-
+   * @name Functions to support hp adaptivity
```
Comment (marcfehling) [inline include/deal.II/fe/fe.h:1315]:
```suggestion
   * @name Functions to support hp-adaptivity
```

File: include/deal.II/hp/fe_collection.h [hunk 3]
```diff
@@ -260,6 +268,18 @@ namespace hp
     unsigned int
     max_dofs_per_cell() const;
 
+    /**
+     * @}
+     */
+
+    /**
+     * @name Functions to support hp adaptivity
```
Comment (marcfehling) [inline include/deal.II/hp/fe_collection.h:276]:
```suggestion
     * @name Functions to support hp-adaptivity
```

**Review by marcfehling — CHANGES_REQUESTED** [review]
We decided on #11216 that we use a hyphen when referring to hp-methods.

I guess the standalone `hp-` occurrences are remnants of my poorly written script for #11216 :)

File: include/deal.II/hp/fe_collection.h [hunk 4]
```diff
@@ -260,6 +268,18 @@ namespace hp
...
+     * @{
+     */
+
+    /**
+     * @{
+     */
```
Comment (marcfehling) [inline include/deal.II/hp/fe_collection.h:282]:
```suggestion
```
Duplicated opening bracket.

**Review by marcfehling — CHANGES_REQUESTED** [review]

**Comment by bangerth** [comment]
Good point, fixed!

File: include/deal.II/hp/fe_collection.h (hunk 4 above)
Comment (bangerth) [reply include/deal.II/hp/fe_collection.h:281]: I fixed that by removing the other opening brace.

**Comment by marcfehling** [comment]
Looks good, but you still missed one occurrence of `hp-adaptivity`

**Comment by bangerth** [comment]
Good grief...

**Review by marcfehling — APPROVED** [review]
Looks good now! Sorry for being pedantic

**Comment by bangerth** [comment]
It's a personality trait I've been known to share :-)

**Pull request merged**